lib/rspec/support/reentrant_mutex.rb

```diff
@@ -25,10 +25,18 @@ def synchronize
         exit
       end
 
+      def locked?
+        @owner != nil
+      end
+
+      def owned?
+        @owner == Thread.current
+      end
```

Comment:
Great, as per @joshcheek's comments on rspec/rspec-core#2340 we still need these methods on our own version of mutex for 1.8.7 too :)